Atomic/sign.py

```diff
@@ -0,0 +1,107 @@
+from . import util
+from . import Atomic
+import os
+import tempfile
+from .atomic import AtomicError
+import re
+
+
+class Sign(Atomic):
+    def sign(self):
+        # TODO
+        # Atomic is run as sudo. Should we work around that?
+
+        if self.args.debug:
+            util.write_out(str(self.args))
+
+        # Verify all the images provided are valid
+        self.check_input_validity()
+
+        signature_path = util.get_signature_path(self.atomic_config)
+        use_def_sig_path = True if signature_path == self.args.output else False
```

Comment:
No, `signature_path` is only the _base_ of a `signature_path/$reponame@$digest/$siquence`; the `-o` option does not override `signature_path` but the final file name, what the code calls `fq_sig_path`.

The location of the default should perhaps be in a utility function, sure, but that won’t resolve my concern that we are mixing strings with two different semantics, especially in this comparison.

(Of course one way to resolve this is to redefine `-o` to specify the base, not the final file.)